src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/EventCounter.cs

```diff
@@ -264,7 +286,8 @@ private void OnEventSourceCommand(object sender, EventCommandEventArgs e)
                 float value;
                 if (e.Arguments.TryGetValue("EventCounterIntervalSec", out valueStr) && float.TryParse(valueStr, out value))
                 {
-                    EnableTimer(value);
+                    lock (this)      // Lock the EventCounterGroup
+                        EnableTimer(value);
```

Comment:
It looks like EnableTimer may end up actually invoking user code, and we're holding a lock.  We're ok with that?  What if that code ends up calling back to one of the method on EventCounter that takes its lock?  Since those locks appear to be holdable while calling this, that would seem to set us up for a cycle that could result in a deadlock.